File: Maximum Gap.cpp
class Solution {
public:
    int maximumGap(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int n = nums.size();
        if (n < 2) return 0;
        int maxx = -1;
        for (int i = 1; i<n; i++) {
            maxx = max(maxx, nums.at(i)-nums.at(i-1));
        }
        return maxx;
    }
};

File: Maximum Gap.py
class Solution:
    def maximumGap(self, nums: List[int]) -> int:
        nums.sort()
        n = len(nums)
        if n < 2:
            return 0
        maxx = -1
        for i in range(1, n):
            maxx = max(maxx, nums[i] - nums[i - 1])
        return maxx

